component/src/main/java/io/siddhi/extension/store/mongodb/MongoDBEventTable.java

```diff
@@ -623,4 +626,262 @@ protected void destroy() {
+        return compiledGroupByJSON.toString();
+    }
+
+    private String getGroupByProjectionString(List<SelectAttributeBuilder> selectAttributeBuilders,
+                                              List<ExpressionBuilder> groupByExpressionBuilders) {
+        StringBuilder compiledProjectionJSON = new StringBuilder();
+        List<MongoSelectExpressionVisitor> getSelectExpressionVisitorList =
+                getSelectAttributesList(selectAttributeBuilders);
+        compiledProjectionJSON.append("{$project:{_id:0,");
+        for (int i = 0; i < getSelectExpressionVisitorList.size(); i++) {
+            MongoSelectExpressionVisitor visitor = getSelectExpressionVisitorList.get(i);
+            String rename = selectAttributeBuilders.get(i).getRename();
+            compiledProjectionJSON.append(rename).append(":1");
+            if (getSelectExpressionVisitorList.indexOf(visitor) == (getSelectExpressionVisitorList.size() - 1)) {
+                compiledProjectionJSON.append("}");
+            } else {
+                compiledProjectionJSON.append(",");
+            }
+        }
+        compiledProjectionJSON.append("}");
+        return compiledProjectionJSON.toString();
+    }
+
+    private String getHavingString(ExpressionBuilder havingExpressionBuilder) {
+        MongoExpressionVisitor visitor = new MongoExpressionVisitor();
+        havingExpressionBuilder.build(visitor);
+        String having = visitor.getHavingCompiledCondition();
+        return "{$match:" + having + "}";
+    }
+
+    private String getOrderByString(List<OrderByAttributeBuilder> orderByAttributeBuilders) {
+        StringBuilder compiledOrderByJSON = new StringBuilder();
+        List<MongoExpressionVisitor> orderByExpressionVisitorList =
+                orderByAttributeBuilders.stream().map((orderByAttributeBuilder -> {
+                    ExpressionBuilder expressionBuilder = orderByAttributeBuilder.getExpressionBuilder();
+                    MongoExpressionVisitor orderByVisitor = new MongoExpressionVisitor();
+                    expressionBuilder.build(orderByVisitor);
+                    return orderByVisitor;
+                })).collect(Collectors.toList());
+        compiledOrderByJSON.append("{$sort:{");
+        for (int i = 0; i < orderByExpressionVisitorList.size(); i++) {
+            MongoExpressionVisitor visitor = orderByExpressionVisitorList.get(i);
+            String order = orderByAttributeBuilders.get(i).getOrder().name();
+            if (visitor.getStreamVarCount() == 0) {
+                compiledOrderByJSON.append(visitor.getCompiledCondition())
+                        .append((order.equalsIgnoreCase("ASC")) ? ":1" : ":-1")
+                        .append((orderByExpressionVisitorList.indexOf(visitor) ==
+                                (orderByExpressionVisitorList.size() - 1)) ? '}' : ',');
+            } else {
+                throw new MongoTableException("The MongoDB Event table does not support order by using " +
```

Comment:
```suggestion
                throw new MongoTableException("The MongoDB Event table does not support 'order by' clause with " +
```
Make similar changes in other logs

Reply:
Fixed in
https://github.com/siddhi-io/siddhi-store-mongodb/pull/56/commits/a270f77b9c9636c73cf908db7b64f418359d589f